app/assets/javascripts/admin/filters/import_date_filter.js.coffee

```diff
@@ -0,0 +1,4 @@
+angular.module("ofn.admin").filter "importDate", ($filter) ->
+  return (products, importDate) ->
+    return products if importDate == "0"
+    $filter('filter')( products, { import_date: importDate }, true )
```

Comment:
I've been looking at why the specs fail on this, and it seems you need to remove the `true` flag from the end of the filter. The inputs being filtered in this case are date strings, which apparently get interpreted differently by angular with this setting. Removing it gets both the failing specs passing again. I also tested manually to double check and it is working as intended.